src/serialization/tokenizer.hpp

```diff
@@ -20,45 +20,91 @@
 
 #include "buffered_istream.hpp"
 
+#include <array>
 #include <istream>
 #include <string>
 
+// use of illegal utf8 character for this purpose was added in a76be7ef1e921dabacd99f16ef440bf9673b8d98
+// added by the preprocessor to allow special handling for #line and #textdomain commands
+// see OUTPUT_SEPARATOR in preprocessor.cpp
+constexpr int INLINED_PREPROCESS_DIRECTIVE_CHAR = 254;
```

Comment:
Comment is no longer accurate. Also, something about this constant seems a bit weird… it's a char but it's an int? Maybe there's a reason for that…

Reply:
I assume it has something to do with overflow wrapping, but if I just use `INLINED_PREPROCESS_DIRECTIVE_CHAR` without casting it to `char` then it fails to parse config files - trying to run wesnoth fails to start with an error about an unexpected character in fonts.cfg.

Reply:
Or maybe more likely something to do with the preprocessor appending using a stringstream's `<<` operator while the tokenizer appends with `+=` to a string.

Reply:
Yes, putting an int into a `<<` operator would format it as an int, not a char.